docs(css): Fx116 - Update "offset-position" property syntax (#27726)
mdn/content: files/en-us/web/css/offset-position/index.md

### Values

- `normal`

>   - : Indicates that the element does not have an offset starting position. This is the default value when the {{cssxref("path")}} CSS function is used to specify the value for the {{cssxref("offset-path")}} property. When the {{cssxref("ray")}} CSS function is used to specify the value for the `offset-path` property, the `normal` keyword defaults to `0px, 0px` of the containing block.

Reviewer: ```suggestion
  - : Indicates that the element does not have an offset starting position. This is the default value when the {{cssxref("path")}} CSS function is used to specify the value for the {{cssxref("offset-path")}} property.
```

or possibly just:

```suggestion
  - : Indicates that the element does not have an offset starting position.
```

This is confusing for me. The values generally just list what the `normal` value does/means. So second suggestion.

The second sentence brings in `path()` and `offset-path`, but the reader doesn't know what any of these things mean, and the spec and page say the initial  or default value is `auto`. likely we should not include this second sentence on this page, but rather, explain it in the example, guide, or additional narrative.

The last sentence should be on the `ray()` page,. I we include it here, it should be in the example. (Also, would be `0 0` or "top left" as text, i think, if we did include it)

- `auto`

>   - : Indicates that the offset starting position is the top-left corner of the element's box. This is the default value when `ray()` is used to specify the value for the {{cssxref("offset-path")}} property.

Reviewer: ```suggestion
  - : Indicates that the offset starting position is the top-left corner of the element's box.
```
This is always the default value, so second sentence isn't needed. Includig the ray() "exception" makes it sound like it's an exception. so best to not include this.

Author: You're right. I've just left the part to say that this is the default value.

### Values

- `normal`
  - : Indicates that the element does not have an offset starting position.
- `auto`
  - : Indicates that the offset starting position is the top-left corner of the element's box. This is the default value.
- `<position>`

>   - : A {{cssxref("&lt;position&gt;")}}. A position defines an x/y coordinate, to place an item relative to the edges of an element's box. It can be defined using one to four values. If two non-keyword values are used, the first value represents the horizontal position and the second represents the vertical position. If only one value is specified, the second value is assumed to be `center`. If three or four values are used, the length-percentage values are offsets for the preceding keyword value(s). For more explanation of these value types, see {{cssxref("background-position")}}.

Author: Done

## Examples

### Comparing various offset starting positions

> This example visually compares the initial offset starting position of an element when `ray()` is used to specify a value for the {{cssxref("offset-path")}} property. The `+` text inside the element box is used to depict the element's anchor point.

Reviewer: the ray example should come second as ray() is not supported in browsers yet and the other example works in most browsers

```html hidden

> <div id="wrap">
>   <div class="box">+</div>
>   <div class="box box0">+</div>
> </div>
> 
> <div id="wrap">
>   <div class="box">+</div>
>   <div class="box box1">+</div>
>   <p>offset-position: normal;</p>
> </div>
> 
> <div id="wrap">
>   <div class="box">+</div>
>   <div class="box box2">+</div>
>   <p>offset-position: auto;</p>
> </div>
> 
> <div id="wrap">
>   <div class="box">+</div>
>   <div class="box box3">+</div>
>   <p>offset-position: left top;</p>
> </div>

Reviewer: ```suggestion
<div class="wrap">
  <div class="box">+</div>
  <div class="box box0">+</div>
</div>

<div class="wrap">
  <div class="box">+</div>
  <div class="box box1">+</div>
  <p>offset-position: normal;</p>
</div>

<div class="wrap">
  <div class="box">+</div>
  <div class="box box2">+</div>
  <p>offset-position: auto;</p>
</div>

<div class="wrap">
  <div class="box">+</div>
  <div class="box box3">+</div>
  <p>offset-position: left top;</p>
</div>

<div class="wrap">
  <div class="box">+</div>
  <div class="box box4">+</div>
  <p>offset-position: 25% 75%;</p>
</div>
```

ids must be unique to a page

Author: Of course! I think I started building out the example with an id and switched to use class later and resulted with a copy paste error. Thanks for catching.

```

```css hidden

> #wrap {

Reviewer: ```suggestion
.wrap {
```

  position: relative;
  width: 500px;
  height: 100px;
  border: 1px solid black;
  margin: 0 2em 4em 5em;
  text-align: center;
  line-height: 3em;
}

p {
  font-family: "Courier New", monospace;
  font-size: 0.8em;
  text-align: right;
  padding-right: 10px;
}

.box {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 50px;
  height: 50px;
}
```

```css
.box {
  background-color: green;
  border-top: 5px solid red;
  position: absolute;
  top: 20px;
  left: 20px;
  opacity: 20%;
}

.box0 {
  opacity: 100%;
}

.box1 {
  offset-position: normal;
  offset-path: ray(0deg);
  opacity: 100%;
}

.box2 {
  offset-position: auto;
  offset-path: ray(0deg);
  opacity: 100%;
}

.box3 {
  offset-position: left top;
  offset-path: ray(0deg);
  opacity: 100%;

> }

Reviewer: ```suggestion
}

.box4 {
  offset-position: 25% 75%;
  offset-path: ray(0deg);
  opacity: 100%;
}
```

Reviewer: adding a non-default value. This example seemed like a ray() example, adding a different value makes it more relevant to offset-position

  width: 40px;
  height: 40px;
  background: cyan;
  position: absolute;
  left: 20px;

>   top: 20px;

Reviewer: on lines 81 and 84, if you make it `0%, 10%` and `90%, 100%` respectively, it will take a "break" at the ends so users can see the effect (or lack thereof in my case) of off-set position.

I am still trying to figure out why it's not working for me....

>   top: 20px;

Author: > That said, I can't get this property to work in the playground.
:) Glad I could show you a demo of the feature working in Fx116 in our 1:1

> if you make it `0%, 10%` and `90%, 100%` respectively
I updated this to `0%, 20%` and `80%, 100%` as you explained in the 1:1

>   top: 20px;

Author: > You could change it to 20px 20px maybe?

So there is a problem with `offset-path: path()`; `path()` currently is not respecting any of the other values for `offset-position`. I tried changing values for `offset-position` but the motion is always starting from top left. I thought at least `normal` would work. In fact, there is no effect of even removing `offset-position` altogether. I am leaving the value `left top` for now (I know this is misleading but it best matches where the path is starting now). I'll raise it as a bug.

}

@keyframes move {
  0%,20% {
    offset-distance: 0%;
  }
  80%,100% {
    offset-distance: 100%;
  }
}
```

#### Result

{{EmbedLiveSample('Setting_initial_offset_position_for_an_offset-path', '100%', 200)}}

### Comparing various offset starting positions

> This example visually compares the initial offset starting position of an element when `ray()` is used to specify a value for the {{cssxref("offset-path")}} property. The `+` text inside the element box is used to depict the element's anchor point.

Reviewer: ```suggestion
This example visually compares the initial offset starting position of an element when `ray()` is used to specify a value for the {{cssxref("offset-path")}} property. The number included inside the element box is used both to indicate which CSS block applied to which box and to depict the element's anchor point.
```

Author: Thanks for catching this change 👍. I've update it to:

> The number inside the element box indicates the element to which CSS is applied as well as the element's anchor point.

}
```

#### Result

{{EmbedLiveSample('Setting_initial_offset_position_for_an_offset-path', '100%', 200)}}

### Comparing various offset starting positions

This example visually compares the initial offset starting position of an element when `ray()` is used to specify a value for the {{cssxref("offset-path")}} property. The `+` text inside the element box is used to depict the element's anchor point.

```html hidden
<div class="wrap">
  <div class="box">+</div>
  <div class="box box0">+</div>
</div>

<div class="wrap">
  <div class="box">+</div>
  <div class="box box1">+</div>

>   <p>offset-position: normal;</p>

Reviewer: Either number the boxes or write out both CSS properties that impact the element

```
  <p>box1: offset-position: normal;</p>
```

or

```
  <p>offset-position: normal;<br/>offset-path: ray(0deg);</p>
```

Author: Numbered the boxes and also displaying `ray()` values now along with each box

</div>

<div class="wrap">
  <div class="box">+</div>
  <div class="box box2">+</div>
  <p>offset-position: auto;</p>
</div>

<div class="wrap">
  <div class="box">+</div>
  <div class="box box3">+</div>
  <p>offset-position: left top;</p>
</div>

<div class="wrap">
  <div class="box">+</div>
  <div class="box box4">+</div>
  <p>offset-position: 30% 70%;</p>
</div>
```

```css hidden
.wrap {
  position: relative;
  width: 500px;
  height: 100px;
  border: 1px solid black;
  margin: 0 2em 4em 5em;
  text-align: center;
  line-height: 3em;
}

p {
  font-family: "Courier New", monospace;
  font-size: 0.8em;
  text-align: right;
  padding-right: 10px;
}

.box {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 50px;
  height: 50px;

> }

Author: Done. Thanks for the tip 👍

```

```css
.box {
  background-color: green;
  border-top: 5px solid red;
  position: absolute;
  top: 20px;
  left: 20px;
  opacity: 20%;
}

.box0 {

>   opacity: 100%;

Reviewer: add either a offset path or an offset position to this example, to show the effect (or lack thereof) of one without the other

Author: Added `offset-position: normal;`

}

.box1 {
  offset-position: normal;
  offset-path: ray(0deg);
  opacity: 100%;
}

.box2 {
  offset-position: auto;
  offset-path: ray(0deg);
  opacity: 100%;
}

.box3 {
  offset-position: left top;
  offset-path: ray(0deg);
  opacity: 100%;
}

.box4 {
  offset-position: 30% 70%;

>   offset-path: ray(180deg);

Reviewer: They all look the same to me, except this one is rotated. I am thinking it might be good to give them all the same ray, of maybe 10deg, or if a ray change makes a huge difference, duplicate box4, with the same non-default OPos but very different Opath

Author: I kept the ray angle as `0deg` because it is the easiest to understand (it is for me :) ) without adding any more variables to see what `offset-position` is doing. I changed the angle for the last one to `120deg` so there is some notion of an angle instead of just a flip.

I've also now added `background-clip` and `dashed` border to fix accessibility issues you illustrated in our 1:1.
